plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -935,9 +944,91 @@ public Optional<ConnectorTableHandle> applyDelete(ConnectorSession session, Conn
     }
 
     @Override
-    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle, RetryMode retryMode)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        if (table.getFormatVersion() < 2) {
+            throw new TrinoException(NOT_SUPPORTED, "Iceberg table updates require at least format version 2");
+        }
+        String deleteMode = table.getStorageProperties().get(DELETE_MODE);
+        if (!"merge-on-read".equals(deleteMode)) {
```

Comment:
per https://github.com/trinodb/trino/pull/11886/files#:~:text=findepi-,7%20hours%20ago,-Actually%2C%20i%20came , please remove the condition

Reply:
Removed